utility_code/lib_py3/market_data.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import sys
+
+from minecraft.player_dat_format.item import Item
+from minecraft.util.debug_util import NbtPathDebug
+
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "../../../quarry"))
+from quarry.types import nbt
+
+class MarketData(NbtPathDebug):
+    """Single class to support both ItemToID and IdToItem maps"""
+
+    def __init__(self, json_data=None, json_path=None, item_to_id=True):
+        """Load from json.
+
+        Must be saved from wherever the json was loaded from for changes to apply.
+
+        item_to_id = True: This is an item -> id map file.
+        item_to_id = False: This is an id -> item map file.
+
+        Since it's the same format for both, just with reversed keys and values, we reuse most of the code for both
+        """
+        self._path = json_path
+        self._data = json_data
+        self._item_to_id = item_to_id
+
+        if self._path is not None and self._data is None:
+            with open(self._path, 'r') as fp:
+                self._data = json.load(fp)
+                fp.close()
+
+        if self._data is None:
+            raise ValueError("No json data or path was provided")
+
+        self.nbt_path_init(None, None, self, None)
+
+    def get_debug_str(self):
+        return str(self)
+
+    def save(self, json_path=None):
+        """Save json data.
+
+        Always returns the json data.
+        If loaded from a path, or a path is provided, saves to that path.
+        """
+        if json_path is None:
+            json_path = self._path
```

Comment:
Note that if the json is loaded from other data - like a large json file with lots of json objects in this format - it creates a file for each of these objects. This if statement should *probably* be removed - but I can imagine it being used temporarily for debugging.